internal/storage/image.go

```diff
@@ -398,6 +404,75 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, configImage, userSpecifiedImage string) (bool, error) {
+	// Resolve the config.Image.Image into a specific immutable c/storage image
+	configImageRef, err := istorage.Transport.ParseStoreReference(svc.store, configImage)
```

Comment:
(I’m not immediately sure this works… if  https://github.com/cri-o/cri-o/pull/7046#discussion_r1228433790 is correct, the input might not always be an image ID.

OTOH if it is either an image ID or a RepoDigests value, and _nothing else_, the digest should fix a specific image, assuming no multi-arch shenanigans.

Ideally it would be possible to turn a `storage.Transport` `ImageReference` directly into a specific `*store.Image`; `storageTransport.GetStoreImage` seems intended to do that but it might not be quite good enough right now. So we might need another function there.

---

Ideally, at some point the input would be resolved into a `*storage.Image` or an image ID, and then create a reference using `ParseStoreReference("@"+id)` == `NewStoreReference(…, nil, id)`.